internal/provider/common_cert.go

```diff
@@ -386,5 +386,12 @@ func createSubjectDistinguishedNames(ctx context.Context, subject certificateSub
 		result.SerialNumber = subject.SerialNumber.ValueString()
 	}
 
+	if !subject.EmailAddress.IsNull() && !subject.EmailAddress.IsUnknown() {
+		result.ExtraNames = append(result.ExtraNames, pkix.AttributeTypeAndValue{
+			Type:  []int{1, 2, 840, 113549, 1, 9, 1},
+			Value: subject.EmailAddress.ValueString(),
```

Comment:
The email address string here will be `UTF-8` as that's the standard encoding for Terraform. [The RFC](https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.6) states this should `IA5String`:
> The attribute value for emailAddress is of type
> IA5String to permit inclusion of the character '@', which is not part
> of the PrintableString character set.

I read through some online discussions and it seems something like this should be a suitable alternative (although I have not tested myself):
```go
Value: asn1.RawValue{
  Tag:   asn1.TagIA5String, 
  Bytes: []byte(subject.EmailAddress.ValueString()),
},
```